addons/account/models/mail_scheduled_message.py

```diff
@@ -0,0 +1,111 @@
+            if scheduled_message.is_note and scheduled_message.from_move_send_wizard and not scheduled_message.env.context.get('from_post'):
+                move_id = self.env['account.move'].browse(scheduled_message.res_id)
+                user = self.env.user
+                create_date = Datetime.context_timestamp(user, Datetime.from_string(scheduled_message.create_date)).strftime('%Y-%m-%d %H:%M:%S')
+                now = Datetime.context_timestamp(user, Datetime.now()).strftime('%Y-%m-%d %H:%M:%S')
+                replace_body = _('scheduled by %(author)s at %(create_date)s is cancelled by %(current_user)s at %(now)s',
+                    author=scheduled_message.author_id.name,
+                    create_date=create_date,
+                    current_user=user.name,
+                    now=now,
+                )
+                body = scheduled_message.body.replace('is scheduled to send', replace_body)
+                move_id.message_post(
+                    author_id=scheduled_message.author_id.id,
+                    body=body,
+                    subtype_xmlid='mail.mt_note',
+                )
+
+    def post_message(self):
+        super(MailScheduledMessage, self.with_context(post_manual=True)).post_message()
+
+    def _post_message(self, raise_exception=True):
+        remained_scheduled_messages = self.env['mail.scheduled.message']
+
+        for scheduled_message in self:
+            payload = scheduled_message.scheduled_send_payload
+            if not (payload and scheduled_message.model == 'account.move'):
+                remained_scheduled_messages += scheduled_message
+                continue
+
+            move_id = self.env['account.move'].browse(scheduled_message.res_id)
+            if not move_id:
+                scheduled_message.with_context(from_post=True).unlink()
+                continue
+
+            sending_settings = {
+                **payload,
+                'pdf_report': self.env['ir.actions.report'].browse(payload['pdf_report']),
+                'mail_template': self.env['mail.template'].browse(payload['mail_template']),
+            }
+            if payload['sending_methods'] and 'email' in payload['sending_methods']:
+                payload_attachments_widget = payload.get('mail_attachments_widget', [])
+                attachments_name_from_payload = [d.get('name') for d in payload_attachments_widget]
+                manual_attachments_widget_data = [
+                    {
+                        'id': attachment.id,
+                        'name': attachment.name,
+                        'mimetype': attachment.mimetype,
+                        'placeholder': False,
+                        'manual': True,
+                    }
+                    for attachment in scheduled_message.attachment_ids
+                    if attachment.name not in attachments_name_from_payload
+                ]
+                sending_settings = {
+                    **sending_settings,
```

Comment:
I know you love multiple layers of methods and dicts combining themselves, but at some point I would advise going through validation methods whitelisting expected attributes, as when reading that code you have no clue what you end up with ^^ .